test/unit/org/apache/cassandra/cql3/validation/entities/VirtualTableTest.java

```diff
@@ -53,52 +57,141 @@
     private static final String VT2_NAME = "vt2";
     private static final String VT3_NAME = "vt3";
 
-    private static class WritableVirtualTable extends AbstractVirtualTable
+    private static class MutableVirtualTable extends AbstractMutableVirtualTable
     {
-        private final ColumnMetadata valueColumn;
-        private final Map<String, Integer> backingMap = new HashMap<>();
+        // <pk1, pk2> -> c1 -> c2 -> <v1, v2>
+        private final Map<Pair<String, String>, SortedMap<String, SortedMap<String, Pair<Integer, Long>>>> backingMap = new ConcurrentHashMap<>();
```

Comment:
NP: this is really hard to follow, at this depth of generics might be worth having a POJO to wrap part of it